lib/ex_unit/lib/ex_unit/formatter.ex

```diff
@@ -303,6 +334,16 @@ defmodule ExUnit.Formatter do
     padding <> Enum.join(reasons, "\n" <> padding) <> "\n"
   end
 
+  defp format_sides(%{left: %ExUnit.Pattern{} = left, right: right}, formatter, inspect) do
+    case format_pattern(left, right, formatter) do
+      {l, r} ->
+        {IO.iodata_to_binary(l), IO.iodata_to_binary(r)}
+
+      nil ->
+        {if_value(left, inspect), if_value(right, inspect)}
```

Comment:
I think `left` is always going to have a value here, so I think `left.binary` can be used directly here.

In addition, I would remove the `else` clause from `format_pattern/3`, and handle`nil`s here because with the current implementation if `pattern_diff/2` returns `nil`, I think `if_value(left, inspect)` will print the `Pattern` struct.